openglyph/include/openglyph/io/mega_filesystem.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <openglyph/io/mega_file.hpp>
+
+#include <cassert>
+#include <filesystem>
+#include <memory>
+#include <unordered_map>
+
+namespace openglyph::io {
+
+/**
+ * @brief A virtual file system for accessing Empire at War .MEG archives.
+ *
+ * The MegaFileSystem class facilitates reading files stored in one or more Petroglyph-style
+ * MegaFile (.MEG) archives used in Star Wars: Empire at War. It searches through multiple
```

Comment:
Don't mention specific game names. This might be used for other games as well.
Also, no need to mention Petroglyph-style. This whole OpenGlyph engine is to support Petroglyph-style stuff :)